hera_cal/redcal.py

```diff
@@ -985,18 +989,41 @@ def redundantly_calibrate(data, reds, freqs=None, times_by_bl=None, fc_conv_crit
         freqs = data.freqs
     if times_by_bl is None:
         times_by_bl = data.times_by_bl
+    data_shape = data[list(data.keys())[0]].shape
+    data_dtype = data[list(data.keys())[0]].dtype
+    gain_keys = sorted(set([utils.split_bl(k)[0] for k in data] + [utils.split_bl(k)[1] for k in data]))
+    vis_keys = [r[0] for r in reds]
 
     # perform firstcal
-    rv['g_firstcal'] = rc.firstcal(data, freqs, maxiter=fc_maxiter, conv_crit=fc_conv_crit)
-    rv['gf_firstcal'] = {ant: np.zeros_like(g, dtype=bool) for ant, g in rv['g_firstcal'].items()}
+    if fc_maxiter > 0:
+        rv['g_firstcal'] = rc.firstcal(data, freqs, maxiter=fc_maxiter, conv_crit=fc_conv_crit, wgts=wgts)
+        rv['gf_firstcal'] = {ant: np.zeros_like(g, dtype=np.bool) for ant, g in rv['g_firstcal'].items()}
```

Comment:
I think you can pull this line out of the conditional. It's equivalent to the version for fc_maxiter == 0.